ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -335,7 +335,7 @@ public CloudBlobMetadata getBlobMetadata(BlobId blobId) throws BlobStorageExcept
    * @throws IllegalStateException on request timeout.
    */
   public AzureCloudDestination.UpdateResponse updateBlobMetadata(BlobId blobId, Map<String, Object> updateFields,
-      CloudUpdateValidator cloudUpdateValidator) throws BlobStorageException {
+      CloudUpdateValidator cloudUpdateValidator) throws Exception {
```

Comment:
Why do we need this change? We are either throwing BlobStorageException here, or RunTimeException.

Reply:
Looks like we could also throw `StoreException` during validation of update method when we call `cloudUpdateValidator.validateUpdate` within this function. Previously, `StoreException` was also incorrectly wrapped in `RunTimeException` which seemed incorrect. Due to that, I had to modify the signature of the method to throw `StoreException` explicitly.